torch/_custom_op/impl.py

```diff
@@ -561,10 +561,16 @@ def validate_schema(schema: FunctionSchema) -> None:
         )
 
 
-def parse_namespace(namespaced_entity: str) -> typing.Tuple[str, str]:
-    names = namespaced_entity.split("::", 1)
+def parse_qualname(qualname: str) -> typing.Tuple[str, str]:
+    names = qualname.split("::", 1)
     if len(names) != 2:
-        raise ValueError(f"Expected there to be a namespace in {namespaced_entity}.")
+        raise ValueError(f"Expected there to be a namespace in {qualname}, i.e. The "
+                         f"operator name should look something like ns::foo")
+    if '.' in names[1]:
+        raise ValueError(f"The torch.custom_ops APIs do not handle non-default "
+                         f"overloads, i.e. operator names with '.' in them. "
+                         f"Please name your operator something like ns::foo. "
+                         f"Got: {qualname}")
```

Comment:
nit: could users be confused about what a "non-default overload" is in this context? My first thought is that that would mean default overload must necessarily end with `.default`, but then as I read further I do see that we clarify that we don't allow `.` in the name, but maybe "do not handle more than one overload" better corresponds to "don't allow periods".